test/serverExplorer.test.ts

```diff
@@ -0,0 +1,301 @@
+        sandbox.stub(serverExplorer.serverStatusEnum, 'get').returns('Stopping');
+        const getChildren = serverExplorer.getChildren(undefined);
+        const getTreeItem = serverExplorer.getTreeItem(serverHandle);
+        serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(serverStatusStub).calledOnce;
+        expect(getStub).calledOnce;
+        expect(getChildren).deep.equals(Array.from([serverHandle]));
+        expect(getTreeItem).deep.equals(serverStopping);
+    });
+});
+
+suite('serverExplorer.addLocation', () => {
+
+    let sandbox: sinon.SinonSandbox;
+    const clientStub: sinon.SinonStubbedInstance<RSPClient> = new RSPClient('somehost', 8080);
+    let serverExplorer: ServersViewTreeDataProvider;
+
+    const findServerBeans = {
+        length: 1,
+        fullVersion: 'version',
+        location: 'path',
+        name: 'EAP',
+        serverAdapterTypeId: 'org.jboss',
+        specificType: 'EAP',
+        version: '7.1'
+    };
+
+    const ServerBeans = {
+        length: 1,
+        fullVersion: 'version',
+        location: 'path',
+        name: 'EAP',
+        serverAdapterTypeId: 'org.jboss',
+        specificType: 'EAP',
+        typeCategory: 'EAP',
+        version: '7.1'
+    };
+
+    const status = {
+        code: 0,
+        message: 'ok',
+        pluginId: 'unknown',
+        severity: 0
+    };
+
+    setup(() => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        sandbox = sinon.createSandbox();
+    });
+
+    teardown(() => {
+        sandbox.restore();
+    });
+
+    test('serverExplorer.addLocation should ask for location of the server and name if server detected in provided location', async () => {
+        const findServerStub = sandbox.stub(clientStub, 'findServerBeans').resolves([ServerBeans]);
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const showOpenDialogStub = sinon.stub(window, 'showOpenDialog').resolves([{fsPath: 'path/path'}]);
+        sandbox.stub(window, 'showInputBox').resolves('eap');
+        sandbox.stub(clientStub, 'createServerAsync').resolves(status);
+        await serverExplorer.addLocation();
+        expect(findServerStub).calledOnce;
+        expect(showOpenDialogStub).calledOnce;
+    });
+
+    test('serverExplorer.addLocation should call client.createServerAsync with detected server bean for location and name provided by user', async () => {
+        const findServerStub = sandbox.stub(clientStub, 'findServerBeans').resolves([findServerBeans]);
+        sandbox.stub(window, 'showInputBox').resolves('eap');
+        sandbox.stub(clientStub, 'createServerAsync').resolves(status);
+        await serverExplorer.addLocation();
+        expect(findServerStub).calledOnce;
+    });
+
+    test('should show message if no server detected in provided location', async () => {
```

Comment:
Does not look right, it should check that window.showInformationMessage was called.